base/src/com/thoughtworks/go/util/OperatingSystem.java

```diff
@@ -12,44 +12,65 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *************************GO-LICENSE-END***********************************/
+ */
 
 package com.thoughtworks.go.util;
 
-/**
- * @understands operating system type
-*/
-public enum OperatingSystem {
-    LINUX("Linux"), OSX("Mac OS X"), SUN_OS("SunOS"), WINDOWS("Windows"), UNKNOWN("Unknown");
-    private String name;
+import org.apache.commons.lang.StringUtils;
 
-    OperatingSystem(String name) {
-        this.name = name;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Properties;
+import java.util.Scanner;
+
+public class OperatingSystem {
+    private static final String OS_FAMILY_NAME = System.getProperty("os.name");
+    public static final String WINDOWS = "Windows";
+    private static String OS_COMPLETE_NAME = detectCompleteName();
+
+    private static String detectCompleteName() {
+        String[] command = {"python", "-c", "import platform;print(platform.linux_distribution())"};
+        try {
+            Process process = Runtime.getRuntime().exec(command);
+            Scanner scanner = new Scanner(process.getInputStream());
+            String line = scanner.nextLine();
+            OS_COMPLETE_NAME = cleanUpPythonOutput(line);
+        } catch (Exception e) {
+            try {
+                OS_COMPLETE_NAME = readFromOsRelease();
+            } catch (Exception ignored) {
+                OS_COMPLETE_NAME = OS_FAMILY_NAME;
+            }
+        }
+        return OS_COMPLETE_NAME;
     }
 
-    public static OperatingSystem fromProperty() {
-        String osName = new SystemEnvironment().getPropertyImpl("os.name");
-        return parseOperatingSystem(osName);
+    private static String readFromOsRelease() throws Exception {
+        FileReader fileReader;
```

Comment:
@aj-jaswanth 
Please this commit here
https://github.com/gocd/gocd/commit/4f7db5908a5163ef4895003dc62be8a63a7e0c6b

I had reverted it

We cannot use StringUtils or anything from apache commons as they are not loaded into the class path when     the agent or server is packaged.